mongo/single_result.go

```diff
@@ -71,6 +88,12 @@ func (sr *SingleResult) setRdrContents() error {
 		return nil
 	case sr.cur != nil:
 		defer sr.cur.Close(context.TODO())
+
+		// Set contents of rdr to cur.Current if non-nil. Otherwise, iterate underlying
+		// cursor.
+		if sr.rdr = sr.cur.Current; sr.rdr != nil {
```

Comment:
Most cursors start with an empty first batch and must be iterated with `Next` to update `Current`. `Cursor`s created with `NewSingleResultFromBytes` start with a pre-loaded `Current`, so we don't need to iterate. This new block shouldn't affect regular `SingleResult`'s rdr-setting logic, but let me know if you think otherwise.